controller/controller_test.go

```diff
@@ -1040,12 +1040,22 @@ func TestControllerConfig(t *testing.T) {
 		t.Errorf("SetBalancer produced unexpected mutation (-want +got)\n%s", diff)
 	}
 
-	// Now that an IP is allocated, removing the IP pool is not allowed.
-	if c.SetPools(l, &config.Pools{ByName: map[string]*config.Pool{}}) != controllers.SyncStateError {
-		t.Fatalf("SetPools that deletes allocated IPs was accepted")
+	// Now that an IP is allocated, removing the IP pool should be accepted.
```

Comment:
nit: removing the IP pool should be accepted even if the IP is allocated, and triggers a reprocess to force the service to get a new IP